src/tests.rs

```diff
@@ -470,3 +470,736 @@ fn add_liquidity_test_2() -> Result<(), ContractError> {
+        },
+    )?;
+    let _ = execute(
+        deps.as_mut(),
+        mock_env(),
+        info.clone(),
+        ExecuteMsg::RegisterJobId {
+            chain_id: Uint256::from(0u8),
+            job: "create_pool".to_string(),
+            job_id: "create_pool_eth".to_string(),
+        },
+    )?;
+    let _ = execute(
+        deps.as_mut(),
+        mock_env(),
+        info.clone(),
+        ExecuteMsg::RegisterJobId {
+            chain_id: Uint256::from(1u8),
+            job: "create_pool".to_string(),
+            job_id: "create_pool_bnb".to_string(),
+        },
+    )?;
+    let _ = execute(
+        deps.as_mut(),
+        mock_env(),
+        info.clone(),
+        ExecuteMsg::CreatePool {
+            chain0_id: Uint256::from(0u8),
+            chain1_id: Uint256::from(1u8),
+            token0: "0x0000000000000000000000000000000000000000".to_string(),
+            token1: "0x1234567890123456789012345678901234567890".to_string(),
+            chain0_init_depositor: "0x0000000000000000000000000000000000000001".to_string(),
+            chain1_init_depositor: "0x0000000000000000000000000000000000000001".to_string(),
+            fee: 3000,
+        },
+    )?;
+    let info = mock_info("tracker0", &[]);
+    let token_amount = 1_000_000_000_000_000_000u128;
+    let _ = execute(
+        deps.as_mut(),
+        mock_env(),
+        info.clone(),
+        ExecuteMsg::AddLiquidity {
+            pool_id: Uint256::from(0u8),
+            chain_id: Uint256::from(0u8),
+            token: "0x0000000000000000000000000000000000000000".to_string(),
+            amount: Uint256::from(token_amount),
+            sender: "0x0000000000000000000000000000000000000001".to_string(),
+            receiver: Addr::unchecked("liquidity_adder"),
+        },
+    )?;
+    let _ = execute(
+        deps.as_mut(),
+        mock_env(),
+        info.clone(),
+        ExecuteMsg::AddLiquidity {
+            pool_id: Uint256::from(0u8),
+            chain_id: Uint256::from(1u8),
+            token: "0x1234567890123456789012345678901234567890".to_string(),
+            amount: Uint256::from(token_amount),
+            sender: "0x0000000000000000000000000000000000000001".to_string(),
+            receiver: Addr::unchecked("liquidity_adder"),
+        },
+    )?;
+    let info = mock_info("liquidity_adder", &[]);
+    let res = execute(
+        deps.as_mut(),
+        mock_env(),
+        info.clone(),
+        ExecuteMsg::RemoveLiquidity {
+            chain0_id: Uint256::from(0u8),
+            chain1_id: Uint256::from(1u8),
+            token0: "0x0000000000000000000000000000000000000000".to_string(),
+            token1: "0x1234567890123456789012345678901234567890".to_string(),
+            receiver0: "receiver0".to_string(),
+            receiver1: "receiver1".to_string(),
+            amount: Uint256::from(token_amount / 2),
+        },
+    )?;
+    assert_eq!(res.messages.len(), 2);
+    Ok(())
+}
+
+fn swap_test() -> Result<(), ContractError> {
```

Comment:
These lint errors are relatively important here, these functions need to be marked `#[test]`